lib/Source.ml

```diff
@@ -209,16 +221,27 @@ let is_long_pexp_open source {Parsetree.pexp_desc; _} =
       contains_token_between source ~from ~upto Parser.IN
   | _ -> false
 
-let is_long_pmod_functor source Parsetree.{pmod_desc; pmod_loc= from; _} =
+let is_long_functor_syntax src ~from = function
+  | Parsetree.Unit -> false
+  | Parsetree.Named ({loc= upto; _}, _) -> (
+      (* before 4.12 the functor keyword is the first token of the functor
+         parameter *)
+      contains_token_between src ~from ~upto Parser.FUNCTOR
+      ||
```

Comment:
Rather than `||`, what do you think of making a different test depending on `Sys.ocaml_version` (which we can expose as a parsed pair in `Parse.parser_version`)? That'll make the seam easier to remove once we require 4.12+.

Reply:
Done, it looks better indeed